### PR TITLE
sumologicexporter: Add support for non-string logs

### PR DESCRIPTION
**Pull request by sumo-drosiek**
Signed-off-by: Dominik Rosiek <drosiek@sumologic.com>

**Description:** <Describe what has changed.>

**Link to tracking Issue:** <Issue number if applicable>

**Testing:** <Describe what testing was performed and which tests were added.>

**Documentation:** <Describe the documentation added.>

### CODE REVIEW

File: exporter/sumologicexporter/fields.go [hunk 1]
```diff
@@ -33,3 +33,12 @@ func (f fields) string() string {
 
 	return strings.Join(rv, ", ")
 }
+
+func (f fields) asInterfaceMap() map[string]interface{} {
+	rv := make(map[string]interface{}, len(f))
```
Comment (kkujawa-sumo) [inline exporter/sumologicexporter/fields.go:38]: I prefer more meaningful names.

File: exporter/sumologicexporter/sender_test.go [hunk 2]
```diff
@@ -126,6 +126,40 @@ func exampleTwoDifferentLogs() []pdata.LogRecord {
...
+	attVal.CopyTo(buffer[0].Body())
+
+	buffer[0].Attributes().InsertString("key1", "value1")
+	buffer[0].Attributes().InsertString("key2", "value2")
+
+	buffer[1] = pdata.NewLogRecord()
+
+	attVal = pdata.NewAttributeValueArray()
+	attArr := attVal.ArrayVal()
+	sv := pdata.NewAttributeValue()
```
Comment (kkujawa-sumo) [inline exporter/sumologicexporter/sender_test.go:148]: Maybe `strVal`?

File: exporter/sumologicexporter/sender_test.go [hunk 3]
```diff
@@ -126,6 +126,40 @@ func exampleTwoDifferentLogs() []pdata.LogRecord {
...
+	sv.SetStringVal("lv2")
+	iv := pdata.NewAttributeValue()
```
Comment (kkujawa-sumo) [inline exporter/sumologicexporter/sender_test.go:150]: Maybe `intVal`?

File: exporter/sumologicexporter/fields.go (hunk 1 above)
Comment (pmalek-sumo) [reply exporter/sumologicexporter/fields.go:38]: It all depends on the context I'd say. In this particular `func` having just 5 lines of code and a single loop copying values I'm not sure if there is a name that would increase the readability.

There are many blogs/presentations on this subject, even by Russ Cox :) 
* https://research.swtch.com/names
* https://talks.golang.org/2014/names.slide

**Review by pmalek-sumo — APPROVED** [review]

File: exporter/sumologicexporter/fields.go (hunk 1 above)
Comment (sumo-drosiek) [reply exporter/sumologicexporter/fields.go:38]: `rv` stands for `returnValue` in this case and this funcion should go away with #15 anyway

**Review by kkujawa-sumo — APPROVED** [review]

**Pull request merged**